Prepare endpoint and service integrations for accepting hacker applications

njhuey opened the pull request:
Create an API endpoint /user/apply to accept a hacker application

- Perform any necessary data validation
- Upload the applicant's resume to Google Drive
- Store the application data in MongoDB
- Send a confirmation email to the applicant

njhuey committed:
created path, added data verification

njhuey committed:
added http route and data verification

njhuey committed:
uploads to google drive, stores user in database, sends confirmation email

taesungh reviewed (changes requested):
Thanks for pulling this together and familiarizing yourself with the other services. We have the concept down but could improve some of the internals.

taesungh committed:
Clean up MongoDB handler and add unit tests

- Make database functions generic (not specific to users)
- Add functions to use find_one and update_one
- Create Enum for database collection names
- Add logging messages for error handling

njhuey committed:
improved internals

njhuey committed:
refactored internals

njhuey committed:
added additional size requirements

taesungh committed:
Refactor resume upload process

taesungh committed:
Refactor sending application confirmation email

- Create `email_handler` for sending templated emails from user data
- Add unit test for SendGrid error handling

taesungh committed:
Add unit tests for user/apply and adjust inputs

- Create unit tests for various application submision scenarios
- Adjust application data field names and restrictions
- Check resume content type

taesungh committed:
Make applicant pronouns a list of strings

- A user may specify multiple sets of pronouns

taesungh committed:
Simplify email handler data handling

- Create `ContactInfo` protocol to use instead of user information

taesungh committed:
Include submission time with application data

taesungh committed:
Rename resume upload w/ applicant name & file hash
